crates/red_knot_python_semantic/src/module_name.rs

```diff
@@ -186,6 +187,45 @@ impl ModuleName {
         self.0.push('.');
         self.0.push_str(other);
     }
+
+    /// Appends a name to `self`
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use ruff_python_ast::name::Name;
+    /// use red_knot_python_semantic::ModuleName;
+    ///
+    /// let mut module_name = ModuleName::new_static("foo").unwrap();
+    /// module_name.append(&Name::new("bar"));
+    /// assert_eq!(&module_name, "foo.bar");
+    /// ```
+    pub fn append(&mut self, name: &Name) {
```

Comment:
Nit: I'd call this `push` similar to `PathBuf::push`, `Vec::push`, `String::push`, ...

Reply:
Yes, good catch!  I had my language wires crossed thinking of Python's `list.append`